### ISSUE TITLE
Attempting to import comtypes.viewobject raises AssetionError

### ISSUE THREAD

**Issue by DManowitz** [enthought/comtypes#446]
When I try to import comtypes.viewobject, I get the following error:

```
  File "<stdin>", line 1, in <module>
  File "C:\Users\manow\miniconda3\envs\py37_ml_TfGpuMain\lib\site-packages\comtypes\viewobject.py", line 26, in <module>
    assert sizeof(tagLOGPALETTE) == 8, sizeof(tagLOGPALETTE)
AssertionError: 12
```

This occurs with comtypes 1.1.14 using Python 3.7 or 3.8

**Comment by junkmd** [comment]
`comtypes.util` is an old module that has not been maintained for a long time, [the last modification being 14 years ago](https://github.com/enthought/comtypes/commits/master/comtypes/viewobject.py).

This package originator, [Thomas Heller has stepped down from Python communities](https://blog.python.org/2011/04/thomas-heller-steps-down-as-ctypes.html) and I do not know his current contact information.

However, from the former change diff, it appears that this was migrated from a `.py` file originally generated in `comtypes/gen`.
The reason I think so is that [the first commit did not import `IUnknown` from `comtypes`, but from `comtypes.gen._00020430_0000_0000_0000_C000_00000000000046_0_2_0`](https://github.com/enthought/comtypes/compare/0a44dee6d55afd5ef51bc13c810e597a61e04fad...0662e2f76c231e436363db1b66ae0aa78a61f22b).
If these were defined from scratch, there would be no need for such roundabout imports.

So perhaps this was working well in the originator's 15 year old environment and is not so now.

I am not sure that removing the `assert` size and alignment checks is the solution.
It may be necessary to change the definitions of various stuffs.

I would like to hear from other participants in this community.

**Comment by vasily-v-ryabov** [comment]
It looks like old asserts are valid for 32-bit Python only. For 64-bit Python we need to add conditional asserts. I did similar job for pywinauto a long time ago (between 2010 and 2014).

@DManowitz do you see such issue in 32-bit Python?

**Comment by vasily-v-ryabov** [comment]
It's because of different `POINTER` size for 32- and 64-bit Python. Maybe using `sizeof(POINTER)` in asserts is much simpler than use of the condition.

**Comment by junkmd** [comment]
@DManowitz 
Are you still interested in this issue?

**Comment by junkmd** [comment]
I agree that the errors encountered, particularly those related to `assert sizeof(...) = ...` and `assert alignment(...) = ...`, are likely due to ["different `POINTER` size for 32- and 64-bit Python"](https://github.com/enthought/comtypes/issues/446#issuecomment-1370074726).
We can make these functionalities work correctly across different architectures by modifying these problematic assertions.

After reviewing some older COM documentation, I've gained a better understanding of the use cases for `IViewObject` and related interfaces.
I've also found that `Shell.Explorer` can successfully `QueryInterface` for `IViewObject`, which allows us to add straightforward tests for `SetAdvise`/`GetAdvise` and other methods.

Similar to how we've enabled previously unusable interfaces and methods in #474, I believe we can make `IViewObject` fully functional and usable.
